src/components/structures/RoomStatusBar.tsx

```diff
@@ -142,6 +146,15 @@ export default class RoomStatusBar extends React.PureComponent<IProps, IState> {
         dis.fire(Action.FocusSendMessageComposer);
     };
 
+    private onRefreshTimelineClick = (): void => {
+        // Empty out the current timeline and re-request it
+        this.props.room.refreshLiveTimeline();
+
+        this.setState({
+            timelineNeedsRefresh: false,
+        });
+    };
```

Comment:
I think the behavior of all of this would be best served by an end-to-end test. It looks like we just [added Cypress recently](https://github.com/matrix-org/matrix-react-sdk/pull/8295) but it's missing all of the utility necessary to complete something like this.

 - Needs a command to create a user behind the scenes and visit Element as the user already signed in
 - Needs commands to setup rooms and send events
 - Needs a way to enable `experimental_features` in the `homeserver.yaml` template
 - Needs an application service registered on the Synapse instance to interact as (the MSC2716 `/batch_send` endpoint is only accessible from a AS token). No extra AS server needed, just the AS token configured.
 - Needs a way to interact with the homserver directly from the application service token to call `/batch_send`

Reply:
Conversation continued at https://github.com/matrix-org/matrix-react-sdk/pull/8354#discussion_r852588155